lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py

```diff
@@ -0,0 +1,213 @@
+    @swiftTest
+    def test_swift_move_function(self):
+        """Check that we properly show variables at various points of the CFG while
+        stepping with the move function.
+        """
+        self.build()
+
+        self.exec_artifact = self.getBuildArtifact(self.exec_name)
+        self.target = self.dbg.CreateTarget(self.exec_artifact)
+        self.assertTrue(self.target, VALID_TARGET)
+
+        self.do_setup_breakpoints()
+
+        self.process = self.target.LaunchSimple(None, None, os.getcwd())
+        threads = lldbutil.get_threads_stopped_at_breakpoint(
+            self.process, self.breakpoints[0])
+        self.assertTrue(len(threads) == 1)
+        self.thread = threads[0]
+
+        self.do_check_copyable_value_test()
+        self.do_check_copyable_var_test()
+        self.do_check_addressonly_value_test()
+        self.do_check_addressonly_var_test()
+        # ccf is conditional control flow
+        self.do_check_copyable_value_ccf_true()
+        self.do_check_copyable_value_ccf_false()
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+        self.exec_name = "a.out"
+
+    def add_breakpoints(self, name, num_breakpoints):
+        pattern = 'Set breakpoint {} here {}'
+        for i in range(num_breakpoints):
+            pat = pattern.format(name, i+1)
+            brk = self.target.BreakpointCreateBySourceRegex(
+                pat, self.main_source_spec)
+            self.assertTrue(brk.GetNumLocations() > 0, VALID_BREAKPOINT)
+            yield brk
+
+    def do_setup_breakpoints(self):
+        self.breakpoints = []
+
+        self.breakpoints.extend(
+            self.add_breakpoints('copyableValueTest', 3))
+        self.breakpoints.extend(
+            self.add_breakpoints('addressOnlyValueTest', 3))
+        self.breakpoints.extend(
+            self.add_breakpoints('copyableVarTest', 4))
+        self.breakpoints.extend(
+            self.add_breakpoints('addressOnlyVarTest', 4))
+        self.breakpoints.extend(
+            self.add_breakpoints('copyableValueCCFTrueTest',
+                                 5))
+        self.breakpoints.extend(
+            self.add_breakpoints('copyableValueCCFFalseTest',
+                                 3))
+
+    def do_check_copyable_value_test(self):
+        frame = self.thread.frames[0]
+        self.assertTrue(frame.IsValid(), "Couldn't get a frame.")
+
+        # We haven't defined varK yet.
+        varK = frame.FindVariable('k')
+
+        self.assertTrue(varK.value == None, "varK initialized too early?!")
+
+        # Go to break point 2. k should be valid.
+        self.runCmd('continue')
+        self.assertTrue(varK.unsigned > 0, "varK not initialized?!")
+
+        # Go to breakpoint 3. k should no longer be valid.
+        self.runCmd('continue')
+        self.assertTrue(varK.value == None, "K is live but was moved?!")
```

Comment:
`self.assertEqual()` will print the value the the comparison fails for easier debugging.

see `lldb/third_party/Python/module/unittest2/unittest2/case.py` for a full list